packages/markdown-docx/src/OOXMLTransformer/index.js

```diff
@@ -58,171 +87,178 @@ class OoxmlTransformer {
+                value: nodeInformation.value || 'Not provided',
+                elementType: nodeInformation.elementType || 'Not a template variable',
+                name: nodeInformation.name || 'No name available',
+            };
+        } else {
+            obj = {
+                $class: `${NS_PREFIX_CommonMarkModel}Text`,
+                text: nodeInformation.value,
+            };
         }
-        return null;
+        for (let i = nodeInformation.properties.length - 1; i >= 0; i--) {
+            obj = {
+                $class: nodeInformation.properties[i],
+                nodes: [obj],
+            };
+        }
+        return obj;
     }
 
     /**
-     * Checks if the element is a line break or not
+     * Creates a pargraph, heading or block CiceroMark Node
      *
-     * @param {object} element the element to be checked
-     * @returns {boolean} whether the element is a line break
+     * @param {object} rootNode Pargraph, heading or any other node
      */
-    isLineBreak(element) {
-        return element.name === 'w:p' && element.elements === undefined;
-    }
+    constructNodes(rootNode) {
+        if (this.JSONXML.length > 0) {
+            let constructedNode;
+            constructedNode = this.construtCiceroMarkNodeJSON(this.JSONXML[0]);
+            rootNode.nodes = [...rootNode.nodes, constructedNode];
 
-    /**
-     * Returns if the element is a heading or not and if it is, an attribute
-     * level is also passed to determine the style of the heading.
-     *
-     * @param {object} headingElement the element to be checked
-     * @returns {object} object of `isHeading` and its level
-     */
-    getHeading(headingElement) {
-        if (headingElement.attributes !== undefined) {
-            const headingLevel = headingElement.attributes['w:val'];
-            if (headingElement.name === 'w:pStyle' && headingLevel.includes('Heading')) {
-                return {
-                    isHeading: true,
-                    level: headingLevel[headingLevel.length - 1],
+            let rootNodesLength = 1;
+            for (let i = 1; i < this.JSONXML.length; i++) {
+                let propertiesPrevious = this.JSONXML[i - 1].properties;
+                let propertiesCurrent = this.JSONXML[i].properties;
+
+                let commonLength = 0;
+                for (let j = 0; j < Math.min(propertiesPrevious.length, propertiesCurrent.length); j++) {
+                    if (propertiesCurrent[j] === propertiesPrevious[j]) {
+                        commonLength++;
+                    } else {
+                        break;
+                    }
+                }
+                let updatedProperties = {
+                    ...this.JSONXML[i],
+                    properties: [...this.JSONXML[i].properties.slice(commonLength)],
                 };
+                constructedNode = this.construtCiceroMarkNodeJSON(updatedProperties);
```

Comment:
Could you explain why are you calling this method twice in the same block? Just wanted to understand the functioning.

Reply:
1. Call for the 1st element in `this.JSONXML`.
2. Call for the remaining elements present in the above array using for loop